src/tests/tracing/eventpipe/common/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,194 @@
+        // uds:/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds://authority/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds:///User/mrx/myport.sock => /User/mrx/myport.sock
+        //
+        // Address in NamedPipe formats:
+        // myport => myport
+        // namedpipe:myport => myport
+        // \\.\pipe\myport => myport (dropping \\.\pipe\ is inline with implemented namedpipe client/server)
+        // namedpipe://./pipe/myport => myport (dropping authority and /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe:/pipe/myport  => myport (dropping /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe://authority/myport => myport
+        // namedpipe:///myport => myport
+        //
+        // PortType: Listen|Connect, default Listen.
+        public static bool TryParse(string config, out IpcEndpointConfig result)
+        {
+            try
+            {
+                result = Parse(config);
+            }
+            catch(Exception)
+            {
+                result = null;
+            }
+            return result != null;
+        }
+
+        public static IpcEndpointConfig Parse(string config)
+        {
+            if (string.IsNullOrEmpty(config))
+                throw new FormatException("Missing IPC endpoint config.");
+
+            string address = "";
+            PortType portType = PortType.Connect;
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : TransportType.UnixDomainSocket;
+#if DIAGNOSTICS_RUNTIME
+            if (OperatingSystem.IsAndroid())
+                transportType = TransportType.TcpSocket;
+#endif
+            if (!string.IsNullOrEmpty(config))
+            {
+                var parts = config.Split(',');
+                if (parts.Length > 2)
+                    throw new FormatException($"Unknow IPC endpoint config format, {config}.");
+
+                if (string.IsNullOrEmpty(parts[0]))
+                    throw new FormatException($"Missing IPC endpoint config address, {config}.");
+
+                portType = PortType.Listen;
+                address = parts[0];
+
+                if (parts.Length == 2)
+                {
+                    if (string.Equals(parts[1], "connect", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Connect;
+                    }
+                    else if (string.Equals(parts[1], "listen", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Listen;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Unknow IPC endpoint config keyword, {parts[1]} in {config}.");
+                    }
+                }
+            }
+
+            if (Uri.TryCreate(address, UriKind.Absolute, out Uri parsedAddress))
```

Comment:
Should we add tcp: schema support as well, that would complete future TCP/IP support for clients.

Reply:
What should the tcp: schema look like? Should it always start with "tcp://"?
For the case of "127.0.0.1:9000" as the address, I thought "Uri.TryCreate" returned false when i tried it once, but I imagine we still want to support `host:port`?
In terms of setting the address, does it parse the address better than `ParseTcpIpEndPoint`

Reply:
Earlier in the comments I proposed that we wouldn't make any changes to Parse(), and thus there would be no string (for now) that parses to a tcp endpoint. My motivation was that if we aren't ready to expose TCP as an end-user feature yet and this is only for testing then there is no need to design a parsing mechanism in advance. Of course if you disagree by all means we can discuss, but I worried we might just be talking past each other and creating some confusion : )

Reply:
OK, now I saw that comment, missed it. So your idea was to move all out to the client and leave library as is:

DiagnosticsClient(new DiagnosticPortIpcEndpoint(new IpcEndpointConfig("127.0.0.1:9000", TransportType.TcpSocket, PortType.Listen)));

and put logic on what transport to use on client side not include any additional support except getting above statement to work through diagnostic client library.

I thought getting the logic into IpcEndPoint was low hanging fruit (since we already put effort into analyze the change) inline with how it would be done going forward and guard those code paths with the DIAGNOSTICS_RUNTIME so it won't be usable from diagnostic repro, but I agree that we could push the logic out of the library and into the test client.

Reply:
I went forward with not modifying `Parse()` and instead directly utilizing the internal `DiagnosticsClient(IpcEndpointConfig config)` constructor instead of the `DiagnosticsClient(IpcEndpoint endpoint)` (since it is more convenient, though not as direct, but if there are any drawbacks I can utilize the endpoint constructor instead).
I also removed the string constructor and added a comment above `IpcEndpointConfig.Address`